designs/2024-per-rule-autofix-configuration/README.md

```diff
@@ -0,0 +1,406 @@
+when creating the `ReportTranslator`. \
+We have the context of the rule and its extended configuration, 
+and we can pass that into `createReportTranslator` as a separate boolean property
+(since the current implementation turns off suggestions when `disableFixes: false`, 
+but we need them if the fixer is disabled). 
+
+Inside `ReportTranslator` we can also convert the fix into a suggestion.
+
+##### Disabling for directives
+
+I think all the work in this regard can be done in the 
+[linter/apply-disabled-directives.js](https://github.com/eslint/eslint/blob/0f49329b4a7f91714f2cd1e9ce532d32202c47f4/lib/linter/apply-disable-directives.js#L466) 
+assuming the full entry of `reportUnusedDisableDirectives` configuration is available.
+
+It's unclear whether suggestions are currently supported for directives - please advise if they are, and where that logic resides.
+
+## Documentation
+
+<!--
+    How will this RFC be documented? Does it need a formal announcement
+    on the ESLint blog to explain the motivation?
+-->
+
+### Autofix on rules
+
+I think it is necessary to demonstrate the use of the full form of the record 
+in the [Configuration Files > Configuring rules](https://eslint.org/docs/latest/use/configure/configuration-files#configuring-rules) section,
+that way we show all the available options literally on one screen.
+
+I think there should be an explanatory section in the [Configure rules](https://eslint.org/docs/latest/use/configure/rules) section, 
+and maybe it should be the first one on the page to show all the available options at once so that the user is not intimidated 
+when they see multiple options on the same screen at the same time like:
+
+```js
+/* eslint eqeqeq: "off", curly: { severity: "error", autofix: false } */
+```
+
+I also believe that there should be a separate "Disabling autofix" section,
+probably located before the [Disabling rules](https://eslint.org/docs/latest/use/configure/rules#disabling-rules) 
+section, so that it can be directly referenced.
+
+### `linterOptions.reportUnusedDisableDirectives`
+
+A note should be added in the
+[Configuration Files > Reporting Unused Disable Directives](https://eslint.org/docs/latest/use/configure/configuration-files#reporting-unused-disable-directives)
+section that autofix behavior can be configured using the extended format.
+
+## Drawbacks
+
+<!--
+    Why should we *not* do this? Consider why adding this into ESLint
+    might not benefit the project or the community. Attempt to think 
+    about any opposing viewpoints that reviewers might bring up. 
+
+    Any change has potential downsides, including increased maintenance
+    burden, incompatibility with other tools, breaking existing user
+    experience, etc. Try to identify as many potential problems with
+    implementing this RFC as possible.
+-->
+
+The primary drawback is the complexity of normalizing configuration formats 
```

Comment:
This is a key detail of implementation that needs fleshing out in this RFC. Figuring out how to efficiently normalize configs so we're not creating a ton of extra objects is important for the implementation.

Reply:
@nzakas The normalization happens in a fairly limited number of places and is easy to implement, it's a pretty straightforward operation. The harder part is identifying and updating all the consumers to work with the normalized object instead of `RuleSeverity | [RuleSeverity, ...unknown[]]`.

Or are you referring to refactoring the code to minimize or even eliminate the use of the non-normalized format altogether, like in [this example](https://github.com/eslint/eslint/blob/0f49329b4a7f91714f2cd1e9ce532d32202c47f4/lib/linter/linter.js#L1086)?

Reply:
The implementation itself needs more thought than I see currently in this RFC. I mentioned this previously: I don't think normalizing all rule configs into an object is the right way to go. It creates A LOT of extra objects in memory that are probably not needed and can slow things down. 

It would be helpful if you could update this RFC with the specific implementation you have in mind. Just saying that certain functions need to be considered doesn't give a clear vision of what you think the implementation will look like.